flink-end-to-end-tests/test-scripts/test_docker_embedded_job.sh

```diff
@@ -30,14 +32,23 @@ export INPUT_PATH=/data/test/input
 export OUTPUT_PATH=/data/test/output
 export FLINK_JOB_ARGUMENTS="--input ${INPUT_PATH}/words --output ${OUTPUT_PATH}/docker_wc_out"
 
-# user inside the container must be able to createto workaround in-container permissions
+build_image() {
+    cd "$DOCKER_MODULE_DIR"
```

Comment:
I would put the `cd "$DOCKER_MODULE_DIR"` and `cd "$END_TO_END_DIR"` around the `if` statement for clarity and also leave the `mkdir -p $OUTPUT_VOLUME` and `chmod 777 $OUTPUT_VOLUME` on top, so that the test fails fast if we do not have the necessary permissions.

In other words, it could look sth like:
```
build_image() {
    ./build.sh --from-local-dist --job-jar ${FLINK_DIR}/examples/batch/WordCount.jar --image-name ${FLINK_DOCKER_IMAGE_NAME}
    local error_code=$?
    return ${error_code}
}

# user inside the container must be able to create files, this is a workaround in-container permissions
mkdir -p $OUTPUT_VOLUME
chmod 777 $OUTPUT_VOLUME

cd "$DOCKER_MODULE_DIR"
if ! retry_times $DOCKER_IMAGE_BUILD_RETRIES ${BUILD_BACKOFF_TIME} build_image; then
    echo "Failed to build docker image. Aborting..."
    exit 1
fi
cd "$END_TO_END_DIR"
```